### PR TITLE
feat(pulsar): add batch support

### PR DESCRIPTION
**Pull request by zamazan4ik**
Resolves https://github.com/vectordotdev/vector/issues/15152

### CODE REVIEW

File: src/sinks/pulsar.rs [hunk 1]
```diff
@@ -77,6 +81,14 @@ pub struct PulsarSinkConfig {
     partition_key_field: Option<String>,
 }
 
+/// Event batching behavior.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Default)]
+pub struct BatchConfig {
+    /// The maximum size of a batch, in events, before it is flushed.
+    pub max_events: Option<u32>,
```
Comment (davidhuie-dd) [inline src/sinks/pulsar.rs:89]: Can we make this `batch_size`? Even though events is the right Vector-specific name for a message, I think that the Pulsar terminology will help people here when they're looking for this setting.

Comment (zamazan4ik) [reply src/sinks/pulsar.rs]: Fixed.

**Comment by zamazan4ik** [comment]
I still do not understand this Cue staff, excuse me. Could you please help me with that? :)

File: website/cue/reference/components/sinks/pulsar.cue [hunk 2]
```diff
@@ -16,6 +16,11 @@ components: sinks: pulsar: {
 		acknowledgements: true
 		healthcheck: enabled: true
 		send: {
+			batch: {
```
Comment (davidhuie-dd) [inline website/cue/reference/components/sinks/pulsar.cue:19]: You should drop the changes to this file. This is meant for configuring certain global docs.

Comment (zamazan4ik) [reply website/cue/reference/components/sinks/pulsar.cue]: Fixed.

P.S. I've always thought that this file relates to the same docs... Thanks for the clarification!

**Comment by zamazan4ik** [comment]
@davidhuie-dd unfortunately, `make generate-component-docs` gives me the following error on my machine:

```
make generate-component-docs
cargo build 
    Finished dev [unoptimized + debuginfo] target(s) in 1.24s
target/debug/vector generate-schema > /tmp/vector-config-schema.json 2>/dev/null
cargo vdev generate component-docs /tmp/vector-config-schema.json \
                
/Users/zamazan4ik/open_source/vector/scripts/generate-component-docs.rb:1735:in `<main>': undefined method `filter_map' for #<Hash:0x00000001248db768> (NoMethodError)
Did you mean?  filter
Error: command: "/Users/zamazan4ik/open_source/vector/scripts/generate-component-docs.rb" "/tmp/vector-config-schema.json"
  failed with exit code: 1
make: *** [generate-component-docs] Error 1
```

Could you please run it on your own and push the result to the PR? Thanks in advance!

**Comment by zamazan4ik** [comment]
Fixed CUE hell (once again).

**Review by davidhuie-dd — APPROVED** [review]
Thanks!

**Comment by jszwedko** [comment]
Stumbling upon this during release preparation. Here this option should just be `batch.max_events` to match our existing batch configuration options and avoid the "stuttering" per https://github.com/vectordotdev/vector/blob/master/docs/specs/configuration.md#option-naming . I'll open a follow-up PR.

**Pull request merged**